cluster-controller/src/main/java/io/strimzi/controller/cluster/Main.java

```diff
@@ -94,4 +106,38 @@ static CompositeFuture run(Vertx vertx, KubernetesClient client, Map<String, Str
         }
         return CompositeFuture.join(futures);
     }
+
+    static Future<Boolean> isOnOpenShift(Vertx vertx, KubernetesClient client)  {
+        URL kubernetesApi = client.getMasterUrl();
+        Future<Boolean> fut = Future.future();
+
+        HttpClientOptions httpClientOptions = new HttpClientOptions();
+        httpClientOptions.setDefaultHost(kubernetesApi.getHost());
+
+        if (kubernetesApi.getPort() == -1) {
+            httpClientOptions.setDefaultPort(kubernetesApi.getDefaultPort());
+        } else {
+            httpClientOptions.setDefaultPort(kubernetesApi.getPort());
+        }
+
+        if (kubernetesApi.getProtocol().equals("https")) {
+            httpClientOptions.setSsl(true);
+            httpClientOptions.setTrustAll(true);
+        }
+
+        HttpClient httpClient = vertx.createHttpClient(httpClientOptions);
+
+        httpClient.getNow("/oapi", res -> {
+            if (res.statusCode() == 200) {
+                log.info("{} returned 200. We should be in OpenShift. It is safe to check 'isAdaptable'", res.request().absoluteURI());
```

Comment:
info, not debug?

Reply:
I used info intentionally. But if you think we can change it to debug.

Reply:
I'm happy with info, I just felt that "It is safe to check 'isAdaptable'" was a bit of an implementation detail.

Reply:
Ok, you convinced me ... moved to debug level.